### PR TITLE
Update to PHPStan 0.12.9 and fix scoping deps

### PR DESCRIPTION
**Pull request by TomasVotruba**
Ref #2749

### CODE REVIEW

**Comment by TomasVotruba** [comment]
@enumag Btw, this was causing the FQN name changes. 

See Github Action fail before fix: https://github.com/rectorphp/rector/pull/2800/checks?check_run_id=428744569#step:5:12

image

File: .github/workflows/test_with_doctrine.yaml [hunk 1]
```diff
@@ -31,4 +31,7 @@ jobs:
                     # do not intall doctrine/orm phpstan, it conflicts with Retor's one
                     composer install -d orm --no-dev
 
+                    # remove phpstan/phpstan to prevent autolaod conflicts
+                    composer remove phpstan/phpstan -d orm
```
Comment (enumag) [inline .github/workflows/test_with_doctrine.yaml]: Since phpstan is phar-ed and prefixed, can it still cause autoload conflicts?

Comment (TomasVotruba) [reply .github/workflows/test_with_doctrine.yaml]: Yes, like this:
https://github.com/rectorphp/rector/pull/2800/checks?check_run_id=430026084

**Comment by enumag** [comment]
> @enumag Btw, this was causing the FQN name changes.

@TomasVotruba huh? not sure what are you trying to tell me... the link isn't showing anything...

**Comment by TomasVotruba** [comment]
@enumag The history was removed now. See the screen

**Pull request merged**